components/automate-ui/src/app/helpers/auth/id-mapper.spec.ts

```diff
@@ -4,11 +4,15 @@ import { IdMapper } from './id-mapper';
 describe('IdMapper', () => {
 
     using([
-      ['my-name', 'my-name', 'mirrors nominal input'],
-      ['-0a5l9z-', '-0a5l9z-', 'supports numbers, letters, and hyphens'],
-      ['my-name!', 'my-name-', 'replaces trailing out-of-range char with hyphen'],
-      ['#$my-name', '--my-name', 'replaces leading out-of-range chars with hyphens'],
-      ['my name', 'my-name', 'replaces space char with hyphen'],
+      ['-0a5l9z-', '-0a5l9z-', 'preserves numbers, lowercase letters, and hyphens'],
+      ['my_name__', 'my_name__', 'preserves underscores'],
+      ['my-name!', 'my-name-', 'replaces trailing unsupported single char with hyphen'],
+      ['my-name$/', 'my-name--', 'replaces trailing unsupported multiple chars with hyphen'],
+      ['my^%<>name', 'my----name', 'replaces middle unsupported chars with hyphen'],
+      ['#my-name', '-my-name', 'replaces leading unsupported single char with hyphens'],
+      [';=@my-name', '---my-name', 'replaces leading unsupported multiple chars with hyphens'],
+      [' my_name  ', '-my_name--', 'replaces leading/trailing spaces with hyphens'],
```

Comment:
Not the job of the IDMapper to trim spaces, so testing it works properly for that case even though calling functions should be doing trimming--see below.